base/server/src/main/java/com/netscape/cmscore/ldapconn/LdapAnonConnFactory.java

```diff
@@ -410,25 +409,38 @@ public synchronized void returnConn(LDAPConnection conn) {
         for (int i = 0; i < mNumConns; i++) {
             if (mConns[i] == anon) {
                 logger.warn("LdapAnonConnFactory: Connection already returned");
+                --mTotal;
+                notifyAll();
+                return;
             }
         }
 
-        // this returned connection might authenticate as someone other than
-        // anonymonus. Reset it to anonymous first before it returns
-        // to the pool.  Do this by calling connect() again on this connection
-        // to avoid doing an explicit anonymous bind
-        try {
-            anon.connect(mConnInfo.getHost(), mConnInfo.getPort());
-        } catch (LDAPException e) {
-            logger.warn("LdapAnonConnFactory: Unable to reauthenticate as anonymous");
-        }
+        if(mNumConns < mMinConns) {
+            // this returned connection might authenticate as someone other than
+            // anonymonus. Reset it to anonymous first before it returns
+            // to the pool.  Do this by calling connect() again on this connection
+            // to avoid doing an explicit anonymous bind
+            try {
+                anon.connect(mConnInfo.getHost(), mConnInfo.getPort());
```

Comment:
This is probably a design issue in the original code. I think recreating the connection in `returnConn()` is kind of defeating the purpose of having a connection pool. Maybe in the future we should revisit this code.